src/schema/comments.ts

```diff
@@ -785,6 +810,34 @@ export const resolvers: IResolvers<any, Context> = {
       });
       return { _: true };
     },
+    reportComment: async (
+      source,
+      { commentId: id, reason, note }: ReportCommentArgs,
+      ctx: Context,
+    ): Promise<GQLEmptyResponse> => {
+      if (!reportCommentReasons.has(reason)) {
+        throw new ValidationError('Reason is invalid');
+      }
+
+      await ctx
+        .getRepository(Comment)
+        .findOneOrFail({ where: { id }, select: ['id'] });
+
+      try {
+        await ctx.getRepository(CommentReport).insert({
+          commentId: id,
+          userId: ctx.userId,
+          reason,
+          note,
+        });
+      } catch (err) {
+        if (err?.code !== TypeOrmError.DUPLICATE_ENTRY) {
+          ctx.log.error({ err }, 'failed to save report to database');
```

Comment:
Shouldn't we throw the error here?

Reply:
We wanted to ignore the error and continue forward. I think was some sort of insert or ignore.

Reply:
It doesn't really makes sense though. Because if the db is not available for example it will not get saved and the user will see that it succeeded. Is it the same for post report as well?
It's ok to not throw it for duplicates but not for all errors

Reply:
Yes, currently works the same. We might need to have a discussion on how we're going to tackle these scenarios moving forward. Would raise it in our next web team meeting.

Reply:
Let's fix them both for now. Should be straightforward. We should throw error when it's not duplicate

Reply:
Awesome. I'll raise a PR quickly.